code/modules/mob/living/living.dm

```diff
@@ -2662,3 +2662,37 @@ GLOBAL_LIST_EMPTY(fire_appearances)
 		physical_cash_total += counted_credit.get_item_credit_value()
 		counted_money += counted_credit
 	return round(physical_cash_total)
+
+/// Returns the approximate power level of this creature
+/mob/living/proc/calculate_power_level()
+	return maxHealth * max((melee_damage_lower + melee_damage_upper) / 2, 0.1)
+
+/// Formats the above as a string
+/mob/living/proc/report_power_level()
+	var/power_level = floor(calculate_power_level())
+	if (power_level >= 9000)
+		return span_boldwarning("[p_Their()] power level is... What?! [power_level]?!?!")
+	return span_notice("[p_Their()] power level is [power_level].")
+
+#define DAMAGE_BOOST 2
+#define DAMAGE_MULT 0.1
+#define SPEED_BOOST 0.1
+#define HEALTH_BOOST 5
+
+/// Generally get stronger, as a saiyan would (or weaker if we pass a negative multiplier)
+/mob/living/proc/saiyan_boost(multiplier = 1)
+	maxHealth += HEALTH_BOOST * multiplier // Fuck knows if this actually does anything
+	var/datum/action/cooldown/mob_cooldown/ki_blast/blast = locate() in actions
+	if (!isnull(blast))
+		blast.damage_modifier += DAMAGE_MULT * multiplier
```

Comment:
do you have to dismiss and resummon your "gun" to update this?

Reply:
Yes but as you would have to enter hard crit in between you would do this automatically via dropping them